clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -189,21 +189,29 @@ protected abstract void onJoinComplete(int generation,
      * Block until the coordinator for this group is known and is ready to receive requests.
      */
     public synchronized void ensureCoordinatorReady() {
-        while (coordinatorUnknown()) {
+        ensureCoordinatorReady(Long.MAX_VALUE);
+    }
+
+    public synchronized void ensureCoordinatorReady(long timeoutMs) {
+        long remainingMs = timeoutMs;
+        long startTimeMs = time.milliseconds();
+        while (coordinatorUnknown() && remainingMs > 0) {
```

Comment:
Hmm, seems like you forgot to remove the `remainingMs` in the `while` condition.